fs/operations.go

```diff
@@ -646,15 +646,35 @@ func Check(fdst, fsrc Fs) error {
 // Lists in parallel which may get them out of order
 func ListFn(f Fs, fn func(Object)) error {
 	list := NewLister().SetFilter(Config.Filter).SetLevel(Config.MaxDepth).Start(f, "")
+
+	var outerErr error
+	var outerErrMut sync.Mutex
+
 	var wg sync.WaitGroup
 	wg.Add(Config.Checkers)
+
 	for i := 0; i < Config.Checkers; i++ {
 		go func() {
 			defer wg.Done()
 			for {
+				// If an error has been set somewhere then simply return from
+				// work.
+				outerErrMut.Lock()
```

Comment:
Whoops, yep. Absolutely right. I added this at the last second and that's what I get.